pkg/rpc/connect.go

```diff
@@ -0,0 +1,211 @@
+package rpc
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	curppb "github.com/xline-kv/go-xline/api/gen/curp"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+// Server Id
+type serverId = uint64
+
+// The connection struct to hold the real rpc connections,
+// it may failed to Connect, but it also retries the next time
+type Connect struct {
+	// Server serverId
+	serverId serverId
+	// The rpc connection
+	RpcConnect []*grpc.ClientConn
```

Comment:
We  could use a custom [load balancer](https://github.com/grpc/grpc-go/blob/master/examples/features/load_balancing/README.md). 
It can be included in the subsequent pull request.